src/storage-types/src/sources.rs

```diff
@@ -172,19 +173,22 @@ impl<S: Debug + Eq + PartialEq + AlterCompatible> AlterCompatible for IngestionD
                                 SourceExport {
                                     output_index: _,
                                     storage_metadata: l_metadata,
+                                    arity: l_arity,
                                 },
                             ),
                             (
                                 _,
                                 SourceExport {
                                     output_index: _,
                                     storage_metadata: r_metadata,
+                                    arity: r_arity,
                                 },
                             ),
                         ) => {
                             // the output index may change, but the table's metadata
                             // may not
-                            l_metadata.alter_compatible(id, r_metadata).is_ok()
+                            l_arity == r_arity
+                                && l_metadata.alter_compatible(id, r_metadata).is_ok()
```

Comment:
Possibly untested (if coverage data is accurate).

Reply:
Very surprising––every call to `ALTER SOURCE...ADD SUBSOURCE` should hit this. I'll take a look into it tomorrow.

Reply:
Thanks for checking it. It is absolutely possible that it is covered but coverage data is off (due to ongoing issues in the coverage collection or tests that failed in the coverage run).